src/dvc_objects/fs/generic.py

```diff
@@ -64,7 +67,9 @@ def _try_links(
             return copy(from_fs, from_path, to_fs, to_path, callback=callback)
 
         try:
-            return _link(link, from_fs, from_path, to_fs, to_path)
+            _link(link, from_fs, from_path, to_fs, to_path)
+            callback.relative_update()
```

Comment:
`transfer` does not do bulk transfers, so it's the responsibility of the caller to do a `relative_update` unless I am missing something. So the callback passed is for bytes level.

Reply:
IMO it made more sense to have the transfer callback operate on the file level so that the callback is always updated when we use `transfer`, vs only when `copy()` is invoked. (And then `transfer` is responsible for branching the callback to the bytes level when `copy` is actually needed)

But I don't feel too strongly about this either way, if we want to revert this change and then do all of the callback branching in all of the dvc-data checkout/save methods that do batch transfers that's fine with me.

Reply:
This does prevent us from relying on progress bar laziness to not show progressbar on `link`, and only on `copy`. But on the other hand, it feels wrong to use a parent callback on what is a single-file operation. hmm.

Reply:
transfer should accept batches though. Ran into this during my work. Lazyness should not be on the transfer side, we should show linking too.

Reply:
> transfer should accept batches though

I don't know how that's relevant to the discussion. At this time, this is a single-file operation at this time.

> Lazyness should not be on the transfer side, we should show linking too.

I am not saying we should not. The callback that works on file-level should not be in `transfer`.

Reply:
The problem from my POV is that `transfer/copy` is not the same as `linking`. Unifying two different concepts are going to be hard.